resolwe_runtime_utils.py

```diff
@@ -275,14 +301,224 @@ def checkrc(rc, *args):
     return json.dumps(ret)
 
 
+def export_file(file_path):
+    """Prepend the given parameter with ``export``"""
+
+    if not os.path.isfile(file_path):
+        return error("Referenced file does not exist: '{}'.".format(file_path))
+
+    return "export {}".format(file_path)
+
+
+CHUNK_SIZE = 10000000  # 10 Mbytes
+
+
+class ImportedFormat:
+    """Import destination file format."""
+
+    EXTRACTED = 'extracted'
+    COMPRESSED = 'compressed'
+    BOTH = 'both'
+
+
+def import_file(
+    src,
+    file_name,
+    imported_format=ImportedFormat.BOTH,
+    progress_from=0.0,
+    progress_to=None,
+):
+    """Import file to working directory.
+
+    :param src: Source file path or URL
+    :param file_name: Source file name
+    :param imported_format: Import file format (extracted, compressed or both)
+    :param progress_from: Initial progress value
+    :param progress_to: Final progress value
+    :return: Destination file path (if extracted and compressed, extracted path given)
+    """
+
+    if progress_to is not None:
+        if not isinstance(progress_from, float) or not isinstance(progress_to, float):
+            raise ValueError("progress_from and progress_to must be float")
+
+        if progress_from >= progress_to:
+            raise ValueError("progress_to must be higher than progress_from")
+
+    print("Importing and compressing {}...".format(file_name))
+
+    def importGz():
+        """Import gzipped file.
+
+        The file_name must have .gz extension.
+        """
+        if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
```

Comment:
I have changed the order as suggested, thanks. Moving the file does not give us much advantage as most of our deploys have the upload folder on a different physical hard drive. Furthermore, we plan to remove files from the upload folder in `purge` if the process completed successfully. This will allow for data recovery in case of errors.